### PR TITLE
feat: expose createdAt and updatedAt on model

### LINKED ISSUE #401: Is it possible to access createdAt and updatedAt?
** Which Category is your question related to? **
amplify schema and DynamoDB

** What AWS Services are you utilizing? **
amplify-js, amplify-cli, AppSync, DynamoDb

** Provide additional details e.g. code snippets **

I've noticed that the Dynamo DB fields `createdAt` and `updatedAt` are automatically generated when I `amplify push` my graphql schema, but I can't figure out a way to query these fields. Is it possible, or should I add my own custom `createdAt` and `updatedAt` fields? Perhaps with different names? Just wondering what the best way to approach this is. 

Without date (`createdAt` and `updatedAt` automatically generated in DynamoDB):

```
type Category @model @auth(rules: [{allow: owner}]) {
  id: ID!
  name: String!
  products: [Product] @connection(name: "CategoryProducts")
}
```

Solution 1 (add `createdOn` field, so not to override auto `createdAt` field):

```
type Category @model @auth(rules: [{allow: owner}]) {
  id: ID!
  createdOn: AWSDateTime!
  name: String!
  products: [Product] @connection(name: "CategoryProducts")
}
```

Solution 2 (override `createdAt` with custom field of same name):
```
type Category @model @auth(rules: [{allow: owner}]) {
  id: ID!
  createdAt: AWSDateTime!
  name: String!
  products: [Product] @connection(name: "CategoryProducts")
}
```

### PR DESCRIPTION
**Pull request by yuth**
Expose createdAt and updatedAt automatically in the transformed schema. Added capability to change the createdAt and updatedAt field names using timestamps

```graphql
type Post @model(timestamps:{createAt: "createdOn", updateAt: "updatedOn"}) {
  id: ID!
  title: String!
  tags: [String!]!
}
```

The above schema will generate Post with `createdOn` and `updatedOn` fields as shown

```graphql
type Post {
  id: ID!
  title: String!
  tags: [String!]!
  createdOn: AWSDateTime!
  updatedOn: AWSDateTime!
}
```
fix #401
doc: https://github.com/aws-amplify/docs/pull/1678

### CODE REVIEW

**Comment by tilk48** [comment]
Hi, will the new fields be usable as parameters in the `@key` directive?

**Comment by yuth** [comment]
> Hi, will the new fields be usable as parameters in the `@key` directive?

In order to use these new fields, they have to be exposed in the type. But you will not have to pass it as input for `create` and `update` mutation

```gql
type Post @model @key(name: "sortByUpdatedAt", fields: ["id", "updatedAt"]) {
  id: ID!
  title: String!
  createdAt: AWSDateTime! # non-nullable field in schema
  updatedAt: AWSDateTime! # non-nullable field schema
}
```

This would generate `CreatePostInput` and `UpdatePostInput` as follows with both `createdAt` and `updatedAt` as nullable field

```gql
input CreatePostInput {
  id: ID
  title: String!
  createdAt: AWSDateTime # this field is nullable and wil be populated automatically if not passed
  updatedAt: AWSDateTime # this field is nullable and wil be populated automatically if not passed
}

input UpdatePostInput {
  id: ID!
  title: String
  createdAt: AWSDateTime # this field is nullable and wil be populated automatically if not passed
  updatedAt: AWSDateTime # this field is nullable and wil be populated automatically if not passed
}
```

File: packages/graphql-dynamodb-transformer/src/definitions.ts [hunk 1]
```diff
@@ -144,16 +156,14 @@ export function makeCreateInputObject(
...
         // ids are always optional. when provided the value is used.
         // when not provided the value is not used.
-        type = {
-          kind: Kind.NAMED_TYPE,
-          name: {
-            kind: Kind.NAME,
-            value: 'ID',
-          },
-        };
+        type = unwrapNonNull(field.type);
```
Comment (attilah) [inline packages/graphql-dynamodb-transformer/src/definitions.ts:166]: is this change related to this PR?

Comment (yuth) [reply packages/graphql-dynamodb-transformer/src/definitions.ts:166]: Yes, this code snippet makes the field to be nullable if they are autogeneratable

File: packages/graphql-dynamodb-transformer/src/DynamoDBModelTransformer.ts [hunk 2]
```diff
@@ -205,6 +248,19 @@ export class DynamoDBModelTransformer extends Transformer {
     let updateFieldNameOverride = undefined;
     let deleteFieldNameOverride = undefined;
 
+    // timestamp fields
```
[the remark this answers is not in the document]
Comment (yuth) [reply packages/graphql-dynamodb-transformer/src/DynamoDBModelTransformer.ts:251]: Done, now the helper function is used by both the fuctions

File: packages/graphql-dynamodb-transformer/src/ModelDirectiveArgs.ts [hunk 3]
```diff
@@ -19,8 +22,35 @@ export interface SubscriptionNameMap {
...
   queries?: QueryNameMap;
   mutations?: MutationNameMap;
   subscriptions?: SubscriptionNameMap;
+  timestamps?: ModelDirectiveTimestampConfiguration;
+}
+
+export function getCreatedAtFieldName(directive: DirectiveNode): string | undefined {
+  const directiveArguments: ModelDirectiveArgs = getDirectiveArguments(directive);
+  const timestamp = directiveArguments.timestamps;
+  if (timestamp === null) return null;
```
[the remark this answers is not in the document]
Comment (yuth) [reply packages/graphql-dynamodb-transformer/src/ModelDirectiveArgs.ts]: Done
